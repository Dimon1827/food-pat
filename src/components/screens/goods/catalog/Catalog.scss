@import '@/assets/styles/variables';

.catalog {
  .catalog__title {
    padding: 27px 0 43px;
    margin-bottom: 40px;


    @media (max-width: $mobile-width) {
      &::after,
      &::before {
        content: "";
        position: absolute;
        bottom: 0;
        left: -20px;
        width: 106%;
        height: 1px;
        background-color: $special-grey;
      }

      &::before {
        top: 0;
      }
    }

    @media (min-width: $tablet-width) {
      padding: 0;
      margin-bottom: 165px;
    }
  }
}

.catalog__container {
  margin-bottom: 24px;

  @media (min-width: $tablet-width) {
    display: grid;
    grid-template-columns: repeat(2, 360px);
    justify-content: center;
    gap: 30px 15px;
    margin-bottom: 70px;
  }

  @media (min-width: $desktop-width) {
    grid-template-columns: repeat(4, 1fr);
    margin-bottom: 76px;
  }
}