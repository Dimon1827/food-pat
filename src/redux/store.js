import { configureStore, combineReducers } from '@reduxjs/toolkit';
import {
  persistReducer,
  persistStore,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import productsReducer from './slice/productsSlice';

const persistConfig = {
  key: 'root',
  version: 2,
  storage,
};

const reducer = combineReducers({
  products: productsReducer,
});

const persistedReducer = persistReducer(persistConfig, reducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
});

export const persistor = persistStore(store);

export default store;
