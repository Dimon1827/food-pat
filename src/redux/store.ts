import { configureStore, combineReducers } from '@reduxjs/toolkit';
import { useDispatch } from 'react-redux';
import {
  persistReducer,
  persistStore,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import productsReducer from './slice/productsSlice';

const persistConfig = {
  key: 'root',
  version: 2,
  storage,
};

const reducer = combineReducers({
  products: productsReducer,
});

const persistedReducer = persistReducer(persistConfig, reducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
});

export type RootState = ReturnType<typeof persistedReducer>;
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const persistor = persistStore(store);
export default store;
