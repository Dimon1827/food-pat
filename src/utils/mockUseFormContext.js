import { useFormContext } from "react-hook-form";

export function mockUseFormContext(errors) {
  useFormContext.mockReturnValue({
    register: jest.fn(),
    formState: {
      errors: errors,
    },
  });
}


