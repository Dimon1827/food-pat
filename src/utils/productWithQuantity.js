const productWithQuantity = (food) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
