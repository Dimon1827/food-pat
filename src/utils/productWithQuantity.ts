import { IProducts } from "redux/slice/productsSlice.types";

interface IproductWithQuantity extends Omit<IProducts, 'isChecked' | 'quantity'> {};

const productWithQuantity = (food: IproductWithQuantity) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
