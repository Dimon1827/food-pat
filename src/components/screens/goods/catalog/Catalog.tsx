import ProductCard from './product-cart/ProductCard';
import productsData from './productsData';
import './Catalog.scss';

function Catalog() {
  return (
    <section className="catalog">
      <div className="container">
        <h1 className="catalog__title title line">Каталог продукции</h1>
        <div className="catalog__container">
          {!!productsData.length &&
            productsData.map((product) => (
              <ProductCard key={product.id} {...product} />
            ))}
        </div>
      </div>
    </section>
  );
}

export default Catalog;
