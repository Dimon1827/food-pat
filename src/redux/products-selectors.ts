import { RootState } from "./store";

export const selectProducts = (state: RootState) => state.products.products;
export const selectTotalPrice = (state: RootState) => state.products.totalPrice;
export const selectTotalQuantity = (state: RootState) => state.products.totalQuantity;
