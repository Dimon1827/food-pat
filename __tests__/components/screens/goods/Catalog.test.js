import { renderWithRouter } from '@/utils/renderWithRouter';
import { screen } from '@testing-library/react';
import Catalog from '@/components/screens/goods/catalog/Catalog';
import * as store from '@/redux/store';

const mockedDispatch = jest.spyOn(store, 'useAppDispatch');


describe('Catalog', () => {
  it('Catalog renders correctly', () => {
    mockedDispatch.mockReturnValue(jest.fn());
    renderWithRouter(<Catalog />);
    expect(screen.getByTestId('catalog')).toMatchSnapshot();
  });
});
