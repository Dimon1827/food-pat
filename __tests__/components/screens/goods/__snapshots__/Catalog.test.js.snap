// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog Catalog renders correctly 1`] = `
<section
  class="catalog"
  data-testid="catalog"
>
  <div
    class="container"
  >
    <h1
      class="catalog__title title line"
    >
      Каталог продукции
    </h1>
    <div
      class="catalog__container"
    >
      <div
        class="product-card catalog__item"
        data-testid="product-card"
      >
        <a
          href="/Овощной салат"
        >
          <picture>
            <source
              height="240"
              srcset="/assets/images/product-image-1.webp"
              type="image/webp"
              width="280"
            />
            <img
              alt="Овощной салат"
              class="product-card__image "
              height="240"
              src="/assets/images/product-image-1.jpg"
              width="280"
            />
          </picture>
        </a>
        <div
          class="product-card__wrapper"
          data-testid="product-card-information"
        >
          <div
            class="product-card__top-wrapper"
          >
            <a
              class="product-card__link"
              href="/Овощной салат"
            >
              <h2
                class="product-card__subtitle subtitle"
              >
                Овощной салат
              </h2>
            </a>
            <button
              class="product-card__button button"
            >
              Купить
            </button>
          </div>
          <p
            class="product-card__text"
          >
            Ингредиенты:
          </p>
          <p
            class="product-card__ingredients"
          >
            Помидоры, зелень, перец, баклажаны
          </p>
          <p
            class="product-card__text"
          >
            Цена:
          </p>
          <span
            class="food__definition food__price"
          >
            700
             ₽
          </span>
        </div>
      </div>
      <div
        class="product-card catalog__item"
        data-testid="product-card"
      >
        <a
          href="/Картошка с мясом"
        >
          <picture>
            <source
              height="240"
              srcset="/assets/images/product-image-2.webp"
              type="image/webp"
              width="280"
            />
            <img
              alt="Картошка с мясом"
              class="product-card__image "
              height="240"
              src="/assets/images/product-image-2.jpg"
              width="280"
            />
          </picture>
        </a>
        <div
          class="product-card__wrapper"
          data-testid="product-card-information"
        >
          <div
            class="product-card__top-wrapper"
          >
            <a
              class="product-card__link"
              href="/Картошка с мясом"
            >
              <h2
                class="product-card__subtitle subtitle"
              >
                Картошка с мясом
              </h2>
            </a>
            <button
              class="product-card__button button"
            >
              Купить
            </button>
          </div>
          <p
            class="product-card__text"
          >
            Ингредиенты:
          </p>
          <p
            class="product-card__ingredients"
          >
            Картошка, мясо, зелень, горох, морковь
          </p>
          <p
            class="product-card__text"
          >
            Цена:
          </p>
          <span
            class="food__definition food__price"
          >
            900
             ₽
          </span>
        </div>
      </div>
      <div
        class="product-card catalog__item"
        data-testid="product-card"
      >
        <a
          href="/Диетическая курица"
        >
          <picture>
            <source
              height="240"
              srcset="/assets/images/product-image-3.webp"
              type="image/webp"
              width="280"
            />
            <img
              alt="Диетическая курица"
              class="product-card__image "
              height="240"
              src="/assets/images/product-image-3.jpg"
              width="280"
            />
          </picture>
        </a>
        <div
          class="product-card__wrapper"
          data-testid="product-card-information"
        >
          <div
            class="product-card__top-wrapper"
          >
            <a
              class="product-card__link"
              href="/Диетическая курица"
            >
              <h2
                class="product-card__subtitle subtitle"
              >
                Диетическая курица
              </h2>
            </a>
            <button
              class="product-card__button button"
            >
              Купить
            </button>
          </div>
          <p
            class="product-card__text"
          >
            Ингредиенты:
          </p>
          <p
            class="product-card__ingredients"
          >
            мясо курицы, помидоры, лимон
          </p>
          <p
            class="product-card__text"
          >
            Цена:
          </p>
          <span
            class="food__definition food__price"
          >
            600
             ₽
          </span>
        </div>
      </div>
      <div
        class="product-card catalog__item"
        data-testid="product-card"
      >
        <a
          href="/Фруктовый салат"
        >
          <picture>
            <source
              height="240"
              srcset="/assets/images/product-image-4.webp"
              type="image/webp"
              width="280"
            />
            <img
              alt="Фруктовый салат"
              class="product-card__image "
              height="240"
              src="/assets/images/product-image-4.jpg"
              width="280"
            />
          </picture>
        </a>
        <div
          class="product-card__wrapper"
          data-testid="product-card-information"
        >
          <div
            class="product-card__top-wrapper"
          >
            <a
              class="product-card__link"
              href="/Фруктовый салат"
            >
              <h2
                class="product-card__subtitle subtitle"
              >
                Фруктовый салат
              </h2>
            </a>
            <button
              class="product-card__button button"
            >
              Купить
            </button>
          </div>
          <p
            class="product-card__text"
          >
            Ингредиенты:
          </p>
          <p
            class="product-card__ingredients"
          >
            Апельсин, банан, киви, черешня
          </p>
          <p
            class="product-card__text"
          >
            Цена:
          </p>
          <span
            class="food__definition food__price"
          >
            500
             ₽
          </span>
        </div>
      </div>
      <div
        class="product-card catalog__item"
        data-testid="product-card"
      >
        <a
          href="/Гречка с горбушой"
        >
          <picture>
            <source
              height="240"
              srcset="/assets/images/product-image-5.webp"
              type="image/webp"
              width="280"
            />
            <img
              alt="Гречка с горбушой"
              class="product-card__image "
              height="240"
              src="/assets/images/product-image-5.jpg"
              width="280"
            />
          </picture>
        </a>
        <div
          class="product-card__wrapper"
          data-testid="product-card-information"
        >
          <div
            class="product-card__top-wrapper"
          >
            <a
              class="product-card__link"
              href="/Гречка с горбушой"
            >
              <h2
                class="product-card__subtitle subtitle"
              >
                Гречка с горбушой
              </h2>
            </a>
            <button
              class="product-card__button button"
            >
              Купить
            </button>
          </div>
          <p
            class="product-card__text"
          >
            Ингредиенты:
          </p>
          <p
            class="product-card__ingredients"
          >
            Гречка, горбуша, помидор, зелень, морковь
          </p>
          <p
            class="product-card__text"
          >
            Цена:
          </p>
          <span
            class="food__definition food__price"
          >
            750
             ₽
          </span>
        </div>
      </div>
      <div
        class="product-card catalog__item"
        data-testid="product-card"
      >
        <a
          href="/Салат с баклажанами"
        >
          <picture>
            <source
              height="240"
              srcset="/assets/images/product-image-6.webp"
              type="image/webp"
              width="280"
            />
            <img
              alt="Салат с баклажанами"
              class="product-card__image "
              height="240"
              src="/assets/images/product-image-6.jpg"
              width="280"
            />
          </picture>
        </a>
        <div
          class="product-card__wrapper"
          data-testid="product-card-information"
        >
          <div
            class="product-card__top-wrapper"
          >
            <a
              class="product-card__link"
              href="/Салат с баклажанами"
            >
              <h2
                class="product-card__subtitle subtitle"
              >
                Салат с баклажанами
              </h2>
            </a>
            <button
              class="product-card__button button"
            >
              Купить
            </button>
          </div>
          <p
            class="product-card__text"
          >
            Ингредиенты:
          </p>
          <p
            class="product-card__ingredients"
          >
            Помидоры, зелень, картошка, баклажаны
          </p>
          <p
            class="product-card__text"
          >
            Цена:
          </p>
          <span
            class="food__definition food__price"
          >
            800
             ₽
          </span>
        </div>
      </div>
      <div
        class="product-card catalog__item"
        data-testid="product-card"
      >
        <a
          href="/Салат с блинами"
        >
          <picture>
            <source
              height="240"
              srcset="/assets/images/product-image-7.webp"
              type="image/webp"
              width="280"
            />
            <img
              alt="Салат с блинами"
              class="product-card__image "
              height="240"
              src="/assets/images/product-image-7.jpg"
              width="280"
            />
          </picture>
        </a>
        <div
          class="product-card__wrapper"
          data-testid="product-card-information"
        >
          <div
            class="product-card__top-wrapper"
          >
            <a
              class="product-card__link"
              href="/Салат с блинами"
            >
              <h2
                class="product-card__subtitle subtitle"
              >
                Салат с блинами
              </h2>
            </a>
            <button
              class="product-card__button button"
            >
              Купить
            </button>
          </div>
          <p
            class="product-card__text"
          >
            Ингредиенты:
          </p>
          <p
            class="product-card__ingredients"
          >
            Зелень, помидор, блины с мясом
          </p>
          <p
            class="product-card__text"
          >
            Цена:
          </p>
          <span
            class="food__definition food__price"
          >
            650
             ₽
          </span>
        </div>
      </div>
    </div>
  </div>
</section>
`;
